slave/base/bin/perun

```diff
@@ -67,7 +67,30 @@ E_DESTINATION_FILE=(209 'Destination file (to diff_update) does not exists or do
 E_PERMISSIONS=(210 'Cannot set permissions')
 E_CHANGE_OWNER=(211 'Cannot set owner')
 
+### declare list of all items in trap on exit
+declare -a on_exit_items
+
 ### Functions
+
+### for purpose of evaluating all on_exit items
+function on_exit()
+{
+    for i in "${on_exit_items[@]}"
+    do
+        eval $i
+    done
+}
+
+### add new on_exit item
+function add_on_exit()
+{
+    local n=${#on_exit_items[*]}
+    on_exit_items[$n]="$*"
+    if [[ $n -eq 0 ]]; then
```

Comment:
Why double '[['?

Reply:
Probably no need for it there. I removed it.